process/smartContract/process.go

```diff
@@ -434,6 +443,15 @@ func (sc *scProcessor) createVMInput(tx data.TransactionHandler) (*vmcommon.VMIn
 
 	vmInput.GasProvided = tx.GetGasLimit() - moveBalanceGasConsume
 
+	if !sc.isAsyncCall {
+		vmInput.CallType = vmcommon.DirectCall
```

Comment:
push vmInput where you need. these 2 flags can be deleted.

Reply:
Fixed (flags deleted, pass `vmInput.CallType` where needed).